homeassistant/components/kulersky/light.py

```diff
@@ -103,21 +81,24 @@ def __init__(self, light: pykulersky.Light):
         self._hs_color = None
         self._brightness = None
         self._white_value = None
-        self._available = True
+        self._available = None
 
     async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
         self.async_on_remove(
-            self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.disconnect)
+            self.hass.bus.async_listen_once(
+                EVENT_HOMEASSISTANT_STOP, self.async_will_remove_from_hass
+            )
```

Comment:
Don't reuse this function for other things. Keep a separate function which you call from both async_will_remove_from_hass.

Reply:
Thanks for the review! That was my first instinct too, but [on my last PR review](https://github.com/home-assistant/core/pull/44430#discussion_r546804360), it was suggested that I [remove the extra function and just call it directly](https://github.com/home-assistant/core/pull/44430/commits/24196a696402f8f0e8bcc28b18a590ca16196dd6). I don't have strong feelings either way.

Reply:
Hmm. Okey. I disagree since the parameters are different. I suppose a lambda could be used as an adapter rather than changing the parameters.

Linting might complain since the overridden method now have different signature compared to base.

Anyway. Its such a minor not so don't mind dropping it.